nvflare/fuel/hci/client/api.py

```diff
@@ -137,11 +345,25 @@ def __init__(
                     raise TypeError(
                         "cmd_modules must be a list of CommandModule, but got element of type {}".format(type(m))
                     )
-        cmd_modules.append(HACommandModule())
+        assert isinstance(service_finder, ServiceFinder), "service_finder should be ServiceFinder but got {}".format(
+            type(service_finder)
+        )
```

Comment:
suggest not using assert so this check will be there with production code

Reply:
changed. But we have many other places have the same issues. Need an overall refactor.